core/components/superboxselect/processors/types/customtable/getlist.class.php

```diff
@@ -0,0 +1,125 @@
+<?php
+/**
+ * Get list resources processor
+ *
+ * @package superboxselect
+ * @subpackage processors
+ */
+
+use TreehillStudio\SuperBoxSelect\Processors\ObjectGetListProcessor;
+
+class SuperboxselectCustomTableGetListProcessor extends ObjectGetListProcessor
+{
+
+    public $classKey = '';
+    public $defaultSortField = 'id';
+    public $defaultSortDirection = 'ASC';
+    private $selectedFields = ['id'];
+
+    /**
+     * @return bool
+     */
+    public function beforeQuery()
+    {
+        // Get Properties
+        $tvid = $this->getProperty('tvid', 0);
+        /** @var modTemplateVar $tv */
+        $tv = $this->modx->getObject('modTemplateVar', $tvid);
+        if ($tv) {
+            $tvProperties = $tv->get('input_properties');
+        } elseif ($this->getProperty('useRequest', 0)) {
+            $tvProperties = $this->getProperties();
```

Comment:
This is quite insecure. Every table can be read by a manager user if the processor is used with `useRequest=1` and the right className. className and selectedFields have to be read from the input properties of the template variable.

Reply:
Good point, thanks! This should do it.